community/collections/src/main/java/org/neo4j/helpers/collection/Iterators.java

```diff
@@ -724,4 +729,44 @@ public static String join( String joinString, Iterator<?> iter )
         }
         return sb.toString();
     }
+
+    /**
+     * Create a stream from the given iterator.
+     * <p>
+     * <b>Note:</b> returned stream needs to be closed via {@link Stream#close()} if the given iterator implements
+     * {@link Resource}.
+     *
+     * @param iterator the iterator to convert to stream
+     * @param <T> the type of elements in the given iterator
+     * @return stream over the iterator elements
+     * @throws NullPointerException when the given stream is {@code null}
+     */
+    public static <T> Stream<T> stream( Iterator<T> iterator )
+    {
+        return stream( iterator, 0 );
+    }
+
+    /**
+     * Create a stream from the given iterator with given characteristics.
+     * <p>
+     * <b>Note:</b> returned stream needs to be closed via {@link Stream#close()} if the given iterator implements
+     * {@link Resource}.
+     *
+     * @param iterator the iterator to convert to stream
+     * @param characteristics the logical OR of characteristics for the underlying {@link Spliterator}
+     * @param <T> the type of elements in the given iterator
+     * @return stream over the iterator elements
+     * @throws NullPointerException when the given iterator is {@code null}
+     */
+    public static <T> Stream<T> stream( Iterator<T> iterator, int characteristics )
+    {
+        Objects.requireNonNull( iterator );
+        Spliterator<T> spliterator = Spliterators.spliteratorUnknownSize( iterator, characteristics );
+        Stream<T> stream = StreamSupport.stream( spliterator, false );
```

Comment:
From the javadoc of [`StreamSupport.stream`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/StreamSupport.html#stream-java.util.Spliterator-boolean-)

> It is strongly recommended the spliterator report a characteristic of IMMUTABLE or CONCURRENT, or be late-binding. Otherwise, stream(java.util.function.Supplier, int, boolean) should be used to reduce the scope of potential interference with the source.

[`Spliterators.spliteratorUnknownSize`](https://docs.oracle.com/javase/8/docs/api/java/util/Spliterators.html#spliteratorUnknownSize-java.util.Iterator-int-) delivers a `Spliterator` that is not `late-binding`.

I am not sure if this is a concern for us, but I want to raise the issue.